File: Kattis_Problem/python/continousmedian.py
# UNSolved TOO SLOW

import pdb


def median(l: list) -> int:
    n = len(l)
    if n%2==0:
        k = n//2
        return (l[k]+l[k-1])//2
    else:
        return l[n//2]

def main():
    T = int(input())
    for _ in range(T):
        N = int(input())
        A = [int(i) for i in input().split()]
        result = 0
        for i in range(len(A)):
            l = A[:i+1]
            l.sort()
            result += median(l)
        print(result)

main()

File: rust/src/continousmedian.rs
mod tools;
use tools::*;

fn median(l: &mut [i32]) -> i32 {
    l.sort();
    let n = l.len();
    if n%2==0 {
        (l[n/2]+l[n/2-1])/2
    } else {
        l[n/2]
    }
}

#[allow(unused_variables, non_snake_case)]
fn main() -> Result<(), Box<dyn error::Error>> {
    let T = input::<i32>()?;
    for _ in 0..T {
        let N = input::<i32>()?;
        let mut A = vec_from_input::<i32>(" ")?;
        let mut result = 0;
        for i in 0..A.len() {
            let l = &mut A[..i+1];
            result += median(&mut l[..]);         
        }
        print!("{}\n",result)
    }
    Ok(())
}